src/knx/bits.cpp

```diff
@@ -124,21 +124,19 @@ uint64_t sixBytesToUInt64(uint8_t* data)
 
 uint16_t crc16Ccitt(uint8_t* input, uint16_t length)
 {
-        uint32_t polynom = 0x1021;
-        uint8_t padded[length+2];
-
-        memcpy(padded, input, length);
-        memset(padded+length, 0x00, 2);
-
-        uint32_t result = 0xffff;
-        for (uint32_t i = 0; i < 8 * (uint32_t)sizeof(padded); i++) {
-            result <<= 1;
-            uint32_t nextBit = (padded[i / 8] >> (7 - (i % 8))) & 0x1;
-            result |= nextBit;
-            if ((result & 0x10000) != 0)
-                result ^= polynom;
-        }
-        return result & 0xffff;
+    uint32_t polynom = 0x1021;
+
+    uint32_t result = 0xffff;
+    for (uint32_t i = 0; i < 8 * ((uint32_t)length + 2); i++)
+    {
+        result <<= 1;
+        uint32_t nextBit;
+        nextBit = ((i / 8) < length) ? ((input[i / 8] >> (7 - (i % 8))) & 0x1) : 0;
```

Comment:
In the original algorithm there were just 2 bytes 0x00 added to the padded field.
The above line just adds the 2 0x00 during access to the array. So the access to the array is just within its size, any access outside of size gives a 0x00.
The main point im my PR is to save RAM. The original data (input) is usually in Flash. If you insist on an external padded buffer, it would require to copy all data to RAM (like in the old algorithm). On a SAMD with a large application there is not enough space in RAM to do so.